### PR TITLE
Remove hyperloglog extension and move hll functions to catalog

### PR DESCRIPTION
**Pull request by oarap**
This PR aims to remove hyperloglog extension and put everything we need in analyze into the system catalog.

### CODE REVIEW

File: src/test/regress/expected/opr_sanity.out [hunk 1]
```diff
@@ -415,9 +415,17 @@ SELECT p1.oid, p1.proname
...
+------+----------------------------------
+ 7158 | hyperloglog_in
+ 7159 | hyperloglog_out
+ 7160 | hyperloglog_comp
+ 7161 | hyperloglog_merge
+ 7162 | hyperloglog_get_estimate
+ 7163 | hyperloglog_add_item_agg_default
+ 7164 | hyperloglog_accum
+ 7165 | hyperloglog_distinct
+(8 rows)
```
Comment (danielgustafsson) [inline src/test/regress/expected/opr_sanity.out]: This test should always return zero rows, adding rows is incorrect. These functions need `DESCRIPTION="<function description>"` in the `WITH ()` block in `pg_proc.sql` to fix that.

File: src/test/regress/expected/opr_sanity.out [hunk 2]
```diff
@@ -851,9 +859,10 @@ WHERE a.aggfnoid = p.oid AND
          NOT binary_coercible(p.proargtypes[2], pfn.proargtypes[3]))
      -- we could carry the check further, but 3 args is enough for now
     );
- aggfnoid | proname | oid | proname 
-----------+---------+-----+---------
-(0 rows)
+ aggfnoid |       proname        | oid  |         proname          
+----------+----------------------+------+--------------------------
+     7165 | hyperloglog_distinct | 7162 | hyperloglog_get_estimate
+(1 row)
```
Comment (danielgustafsson) [inline src/test/regress/expected/opr_sanity.out]: This should also not add a row. Why are the parameter types not binary coercible?

File: src/test/regress/expected/type_sanity.out [hunk 3]
```diff
@@ -64,22 +64,24 @@ WHERE p1.typtype in ('b','e') AND p1.typname NOT LIKE E'\\_%' AND NOT EXISTS
...
- 194 | pg_node_tree
- 210 | smgr
- 705 | unknown
-(3 rows)
+ oid  |        typname        
+------+-----------------------
+  194 | pg_node_tree
+  210 | smgr
+  705 | unknown
+ 7157 | hyperloglog_estimator
```
Comment (danielgustafsson) [inline src/test/regress/expected/type_sanity.out]: Why don't we allow an arraytype?

File: src/test/regress/expected/type_sanity.out [hunk 4]
```diff
@@ -64,22 +64,24 @@ WHERE p1.typtype in ('b','e') AND p1.typname NOT LIKE E'\\_%' AND NOT EXISTS
...
 FROM   pg_type p1 LEFT JOIN pg_type p2 ON (p1.typarray = p2.oid)
 WHERE  p1.typarray <> 0 AND
        (p2.oid IS NULL OR p2.typelem <> p1.oid OR p2.typlen <> -1);
- oid | basetype | arraytype | typelem | typlen 
------+----------+-----------+---------+--------
-(0 rows)
+ oid  |       basetype        | arraytype | typelem | typlen 
+------+-----------------------+-----------+---------+--------
+ 7157 | hyperloglog_estimator | _bytea    |      17 |     -1
+(1 row)
```
Comment (danielgustafsson) [inline src/test/regress/expected/type_sanity.out]: This one and the one below doesn't seem right. What is the intention with hyperloglog array types?

**Comment by oarap** [comment]
@danielgustafsson I applied your comments, could you please have a one more pass.

File: contrib/postgres_hyperloglog/META.json [hunk 5]
```diff
@@ -1,35 +0,0 @@
-{
-   "name": "hyperloglog_estimator",
-   "abstract": "Estimates number of distinct elements in a data set (aggregate and a data type).",
-   "description": "Provides an alternative to COUNT(DISTINCT) aggregate, computing an estimate of number of distinct values, and a data type that may be used within a table (and updated continuously). This implementation is based on HyperLogLog algorithm, an enhancement of LogLog (see the paper 'HyperLogLog: the analysis of near-optimal cardinality estimation algorithm' by Flajolet, Fusy, Gandouet and Meunier, published in 2007).",
-   "version": "2.0.0",
```
Comment (dhanashreek89) [inline contrib/postgres_hyperloglog/META.json]: Should we retain the literature reference by adding it in an appropriate file, `hyperloglog.c` may be?

**Pull request merged**